src/libraries/System.Security.Cryptography.Encoding/tests/PemEncodingTests.cs

```diff
@@ -0,0 +1,548 @@
+            {
+                labels.Add(pem[fields.Label].ToString());
+                pem = pem[fields.Location.End..];
+            }
+
+            Assert.Equal(new string[] { "EC PARAMETERS", "EC PRIVATE KEY" }, labels);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidBase64()
+        {
+            string content = @"
+-----BEGIN TEST-----
+$$$$
+-----END TEST-----
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidLabel()
+        {
+            string content = @"
+-----BEGIN ------
+YmFy
+-----END ------
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_False_Empty()
+        {
+            Assert.False(PemEncoding.TryFind(string.Empty, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PostEbBeforePreEb()
+        {
+            string content = "-----END TEST-----\n-----BEGIN TEST-----\nZm9v";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Theory]
+        [InlineData("\tOOPS")]
+        [InlineData(" OOPS")]
+        [InlineData("-OOPS")]
+        [InlineData("te\x7fst")]
+        [InlineData("te\x19st")]
+        public static void TryFind_False_InvalidLabel(string label)
+        {
+            string content = $"-----BEGIN {label}-----\nZm9v\n-----END {label}-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_InvalidBase64()
+        {
+            string content = "-----BEGIN TEST-----\n$$$$\n-----END TEST-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PrecedingLinesAndSignificantCharsBeforePreeb()
+        {
+            string content = "boop\nbeep-----BEGIN TEST-----\nZm9v\n-----END TEST-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_ContentOnPostEbLine()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n-----END TEST-----boop";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_NoPostEncapBoundary()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n";
+            Assert.False(PemEncoding.TryFind(content, out _));
```

Comment:
I don't know if we want subclassing, or double-duty; but it's probably good to have the tests always throw the same thing at both Find and TryFind, checking for the boolean/exception, and if TryFind succeeds that they give the same data.

With subclassing, it'd be something like

```C#
protected abstract PemFields Find(ReadOnlySpan<char> text);
protected abstract void FailToFind<TException>(ReadOnlySpan<char> text) where TException : Exception;

// (I have it returning the value, in case any multi-read tests wanted to compare two sets against each other)
private PemFields TestSuccess(string content, Range expectedLocation, Range expectedLabel, Range expectedBase64Data, int offset = 0)
{
    PemFields fields = Find(content.AsSpan(offset));
    // helper, as described before
}

public class PemEncodingFindTests_TryFind : PemEncodingFindTests
{
    protected override void FailToFind<TException>(ReadOnlySpan<char> text) where TException : Exception
    {
        Assert.False(PemEncoding.TryFind(text, out PemFields fields));
        Assert.True(fields.Location.Equals(default(Range)));
        ...
    }
}
```